front/components/data_source_view/DataSourceViewSelector.tsx

```diff
@@ -212,8 +356,28 @@ function DataSourceViewSelector({
       ? "partial"
       : "unchecked";
 
-  const onToggleSelectAll = useMemo(() => {
-    return (checked: boolean) => {
+  const keepOnlyOneVaultIfApplicable = useCallback(
```

Comment:
Not sure to understand what we are trying to do here.

Reply:
Added comments.

Reply:
Did you forget to push 🤔